pint_xarray/accessors.py

```diff
@@ -29,6 +31,26 @@
 # TODO type hints
 
 
+def is_dict_like(obj):
+    return hasattr(obj, "keys") and hasattr(obj, "__getitem__")
+
+
+def either_dict_or_kwargs(positional, keywords, method_name):
+    if positional is not None:
+        if not is_dict_like(positional):
+            raise ValueError(
+                f"the first argument to .{method_name} must be a dictionary"
+            )
+        if keywords:
+            raise ValueError(
+                "cannot specify both keyword and positional "
+                f"arguments to .{method_name}"
+            )
+        return positional
+    else:
+        return keywords
```

Comment:
These look awfully similar to the utils of the same name in xarray. Were they implemented independently or reused? If reused, we should probably add reference to that fact (e.g. "Copyright 2014-2019, xarray Developers. Reused under the Apache License, Version 2.0")

Reply:
definitely reused, I'll add a comment stating that.